### ISSUE TITLE
Enable Apple Photos file deletions

### ISSUE THREAD

**Issue by theophanemayaud** [theophanemayaud/video-simili-duplicate-cleaner#11]
After #10 is done : implement proper support for apple photos. Deleting could be done like PhotoSweeper does it : programmatically tell apple photos to move the files not wanted to a specific album for example, and then the user can delete them all simply !

- [x] add support for adding to album in Apple Photos
- [x] check for potential bugs
- [x] Make sure only to check if it is an « original » file ie (I think) in the original folder -> later see #52
- [x] Try to do : reveal in photos behavior instead of open file location -> see #51 
- [x] Enable selection of an Apple Photo Library (can currently be drag and dropped, or scanned if within other selected folder, but not selected directly) -> done see 538d817ff1718f997e8c554ca17f96582180f59f
- [x] add permission for App Store [could not figure it out, waiting for SO](https://stackoverflow.com/questions/70548909/how-to-run-applescript-from-c-in-macos-sandbox-environment-without-entitlement). Finally probably solved by rewriting into Objective C 7d723128358285baa57a342e5bf231f50974fe05
- [x] update readme and all to tell users of new features, and add help message to explain in the app when it happens

**Comment by theophanemayaud** [comment]
It should be done via AppleScript, using the filename of the video on disk within the apple photos library container, which in AppleScript is identified as a media item id, with 033F9BD2-9DB3-4DCD-B281-0714BDC50C48 being the video name, it can be done like this : 
```
tell application "Photos"
	set selMedia to (get media items whose id contains "033F9BD2-9DB3-4DCD-B281-0714BDC50C48")
	if not (album "Script album" exists) then
		make new album named "Script album"
	end if
	add selMedia to album "Script album"
end tell
```

**Comment by theophanemayaud** [comment]
Should have a look at : 
https://wiki.qt.io/Call_an_AppleScript_from_Qt
https://stackoverflow.com/questions/21924932/how-to-run-an-applescript-from-a-sandboxed-application-on-a-mac-os-x
https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/EntitlementKeyReference/Chapters/EnablingAppSandbox.html#//apple_ref/doc/uid/TP40011195-CH4-SW25

**Comment by theophanemayaud** [comment]
We could also get the name of the video (and not its file name on disk) from apple photos. It'd be nice to enable the same name comparison features, and also having a better idea of what file it is !

**Comment by theophanemayaud** [comment]
Note : we should inform the user that only the downloaded files will be scanned, and they need to enable "download all from iCloud on device" not "smart storage" for it to work on the whole library. If not enough space on disk, maybe move library to another disk at least temporarily... 🤷‍♂️

**Comment by theophanemayaud** [comment]
NB : we must check wether the video file is in the originals folder of the library, and if the filename contains underscores after the long id, it seems to indicate that the file is actually a clip to make up a Live Photo video, and not a true video.

**Comment by theophanemayaud** [comment]
For store permissions: look at Photos.sdef file inside Photos.app package, getting inspiration from https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/EntitlementKeyReference/Chapters/EnablingAppSandbox.html#//apple_ref/doc/uid/TP40011195-CH4-SW25 and the Mail.sdef inside Mail.app

**Comment by theophanemayaud** [comment]
closed by 9c47d6a3a79dfb87b9435ea1a73b3d8b3ac81f2a
